src/app/digital/internal/views/components/ORGateAssembler.ts

```diff
@@ -0,0 +1,150 @@
+            .map((id) => this.circuit.getPortByID(id).unwrap())
+            .filter((p) => ((p) && (inputPortGroups.includes(p.group)))).length;
+        
+        return numInputs
+    }
+
+    private assembleQuadCurve(gate: Schema.Component) {
+        const { defaultBorderWidth, selectedBorderColor, defaultBorderColor } = this.options;
+        const transform = this.view.componentTransforms.get(gate.id)!;
+        const selected = this.selections.has(gate.id);
+
+        const numInputs = this.getNumInputs(gate);
+
+        let quadCurves = [];
+
+        const amt = 2 * Math.floor(numInputs / 4) + 1;
+        const [lNumMod, sMod] = (amt === 1) ? ([0.014, 0]) : ([0, 0.012]);
+        for (let i = 0; i < amt; i++) {
+            const d = (i - Math.floor(amt / 2)) * this.size.y;
+            const h = defaultBorderWidth;
+            const l1 = -this.size.y / 2 + lNumMod;
+            const l2 = +this.size.y / 2 - lNumMod;
+
+            const s = this.size.x / 2 - h + sMod;
+            const l = this.size.x / 5 - h;
+
+            let p1 = V(-s, l1 + d);
+            let p2 = V(-s, l2 + d);
+    
+            const c = V(-l, d);
+
+            if (numInputs <= 3) {
+                quadCurves.push(new QuadCurve(
+                    transform.toWorldSpace(p1),
+                    transform.toWorldSpace(p2),
+                    transform.toWorldSpace(c),
+                    {stroke: {color: "black", size: 0.02, lineCap: "round"}}
+                ));
+            } else {
+                quadCurves.push(new QuadCurve(
+                    transform.toWorldSpace(p1),
+                    transform.toWorldSpace(p2),
+                    transform.toWorldSpace(c),
+                    {stroke: {color: "black", size: 0.05, lineCap: "round"}}
+                ));
+            }
+            
+        }
+        return quadCurves;
+    }
+
+    private assembleImage(gate: Schema.Component) {
+        const selected = this.selections.has(gate.id);
+        const tint = (selected ? this.options.selectedFillColor : undefined);
+        return new SVGPrim(this.img, this.size, this.view.componentTransforms.get(gate.id)!, tint);
+    }
+
+    public assemble(gate: Schema.Component, ev: unknown) {
+        const transformChanged = /* use ev to see if our transform changed */ true;
+        const selectionChanged = /* use ev to see if we were de/selected */ true;
+        const portAmtChanged   = /* use ev to see if the number of ports changed */ true;
+
+        if (!transformChanged && !selectionChanged && !portAmtChanged)
+            return;
+
+        if (transformChanged) {
+            // Update transform
+            this.view.componentTransforms.set(gate.id, new Transform(
+                V(gate.props.x ?? 0, gate.props.y ?? 0),
+                this.size,
+                (gate.props.angle ?? 0),
+            ));
+        }
+
+        const numInputs = this.getNumInputs(gate);
+        if (numInputs % 2 === 0)
```

Comment:
You shouldn't need two positioners, you can use `total` in the positioner to check if it's even or odd and return a different position based on that